### PR TITLE
ROX-17275: Replace classic Cluster Health widget with PatternFly cards in System Health

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

TL;DR

Replace classic Cluster Health widget which had 6 sub-widgets with 3 PatternFly cards:
* Cluster status
* Sensor upgrade
* Credential expiration

### Problems

1. Cluster overview omits Unavailable status.
3. Cause and effect for status is unclear.
3. Cards display only most severe status.
4. Color vision accessibility.
5. Tight coupling with clusters for colors and icons (my bad).

### Analysis

Reoganize information following example of declarative configuration to distinguish no errors versus errors:
* Card has head with green check mark, but not body, if non-zero healthy and zero unhealthy and degraded.
* Card has body for request error or for table of counts if zero healthy or any non-zero unhealthy or degraded (which have red or yellow icons).
    * 4 rows make cause and effect relationship clearer in cluster status table.
    * 5 status and total columns make relationship clearer in all tables.

### Solution

1. Move clusters request from DashboardPage to ClustersHealthCards.
2. Begin to separate files in Components to more specific subfolders like ClustersHealth.
3. Delete obsolete components.
4. Rewrite integration tests.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Edited integration tests

### Manual testing

1. Visit classic /main/system-health: see height of Cluster Health widget

    * at 1440px screen width
        classic_1440

    * at 1920px screen width
        classic_1920

2. Visit PatternFly /main/system-health

    * spinner until response from **first** GET /v1/clusters request
        spinner

    * request error message if I temporarily mangle the endpoint address (similarly, Network Error when infra cluster stops)
        requestErrorMessage

    * card header only if **healthy**
        healthy

### Integration testing

1. `yarn cypress-open` in ui

    * systemHealth/clusters.test.js

    * fixture with 7 clusters: see red icons for non-zero counts in unhealthy columns and yellow icons for non-zero counts in degraded columns
      fixture_7_Th

    * fixture with 3 clusters: see Sensor upgrade is healthy, therefore does not need card body with counts table
      fixture_3_Th

    * fixture with 1 cluster that has Uninitialized status: see counts tables because no healthy clusters
        fixture_1_Uninitialized_Th

### CODE REVIEW

File: ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/CredentialExpirationCard.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                        />
+                        <Tbody>
+                            <Tr>
+                                <Td>Clusters</Td>
+                                <TdHealthy counts={counts} dataLabel={dataLabelHealthy} />
+                                <TdUnhealthy counts={counts} dataLabel={dataLabelUnhealthy} />
+                                <TdDegraded counts={counts} dataLabel={dataLabelDegraded} />
+                                <TdUnavailable counts={counts} />
+                                <TdUninitialized counts={counts} />
+                                <TdTotal clusters={clusters} />
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/CredentialExpirationCard.tsx:79]: What are the considerations of passing all the counts to each different type of cell, instead of passing the count each cell needs?

**Review by vjwilson — COMMENTED** [review]
Overall looks good. Some comments and questions for consideration.

File: ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/ClusterStatusCard.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    TdTotal,
+    TdUnavailable,
+    TdUninitialized,
+    TdUnhealthy,
+    TheadClustersHealth,
+} from './ClustersHealthTable';
+
+export type ClusterStatusTableProps = {
+    clusters: Cluster[];
+    isFetching0: boolean;
```
[the remark this answers is not in the document]
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/ClusterStatusCard.tsx:21]: Excellent review from reader viewpoint. Renaming as `isFetchingInitialRequest`

File: ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/ClusterStatusCard.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+              countsSensor !== null &&
+              countsCollector !== null &&
+              countsAdmissionControl !== null &&
+              (counts.HEALTHY === 0 || counts.UNHEALTHY !== 0 || counts.DEGRADED !== 0) ? (
+                <CardBody>
+                    <TableComposable variant="compact">
+                        <TheadClustersHealth />
+                        <Tbody>
+                            <Tr>
+                                <Td>Clusters because any of the following</Td>
```
[the remark this answers is not in the document]
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/ClusterStatusCard.tsx:76]: 1. Indeed, here is a second draft:

    Clusters: overall status
    Clusters: sensor status
    Clusters: collector status
    Clusters: admission control status

    Trying to keep Clusters in common among 3 tables but also careful about quantity. Although there. is one sensor per secured cluster, there are usually several collectors, but there is one collector status per cluster.

2. HTML and PatternFly for the win. `<Th scope="row">…<Th>` does what you suggest!

File: ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/CredentialExpirationCard.tsx (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/SystemHealth/ClustersHealth/CredentialExpirationCard.tsx:79]: Aha, your question pointed me to obsolete code from first draft!

`TdHealthy` component still has `count !== 0 && counts.UNHEALTHY === 0 && counts.DEGRADED === 0` to render green `CheckCircleIcon` but now the table never appears if healthy.

Therefore, I will change from `counts` to `count` prop.

**Review by vjwilson — APPROVED** [review]
